pkg/action/sync.go

```diff
@@ -42,29 +37,44 @@ func (s *Action) sync(ctx context.Context, c *cli.Context, store string) error {
 				continue
 			}
 		}
-
 		numMPs++
-		_ = s.syncMount(ctx, mp)
+		if syncResult, _ := s.syncMount(ctx, mp); syncResult != nil {
+			changes[syncResult.name] = syncResult.changed
+		}
 	}
 	out.Green(ctx, "All done")
 
 	// calculate number of changes entries
-	if l, err := s.Store.Tree(ctx); err == nil {
-		numEntries = len(l.List(0)) - numEntries
+	totalAdded := 0
+	totalRemoved := 0
+	if len(changes) > 0 {
+		out.Print(ctx, "Sync summary:\n")
+	}
+	for k := range changes {
+		if changes[k] > 0 {
+			out.Print(ctx, fmt.Sprintf("Mount %s: Added %d entries", k, changes[k]))
+			totalAdded += changes[k]
+		} else if changes[k] < 0 {
+			out.Print(ctx, fmt.Sprintf("Mount %s: Removed %d entries", k, -1*changes[k]))
+			totalRemoved += -1 * changes[k]
+		} else {
+			out.Print(ctx, fmt.Sprintf("Mount %s: no changes", k))
+		}
 	}
 	diff := ""
-	if numEntries > 0 {
-		diff = fmt.Sprintf(" Added %d entries", numEntries)
-	} else if numEntries < 0 {
-		diff = fmt.Sprintf(" Removed %d entries", -1*numEntries)
+	if totalAdded > 0 {
+		diff += fmt.Sprintf(" Added %d entries", totalAdded)
+	}
+	if totalRemoved > 0 {
+		diff += fmt.Sprintf("\nRemoved %d entries", totalRemoved)
 	}
 	_ = notify.Notify(ctx, "gopass - sync", fmt.Sprintf("Finished. Synced %d remotes.%s", numMPs, diff))
 
 	return nil
 }
 
 // syncMount syncs a single mount
-func (s *Action) syncMount(ctx context.Context, mp string) error {
+func (s *Action) syncMount(ctx context.Context, mp string) (*syncResult, error) {
```

Comment:
Don't use a pointer to syncResult. We don't need to modify the object later.

Reply:
Isn't that convenient so I can return nil?